src/basic/calendarspec.c

```diff
@@ -471,6 +489,20 @@ static int prepend_component(const char **p, bool usec, CalendarComponent **c) {
 
                 if (repeat == 0)
                         return -ERANGE;
+        } else if (e[0] == '.' && e[1] == '.') {
+                e += 2;
+                r = parse_component_decimal(&e, usec, &range_end);
+                if (r < 0)
+                        return r;
+
+                if (value >= range_end)
+                        return -EINVAL;
+
+                for (i = value; i <= range_end; i += usec ? USEC_PER_SEC : 1) {
```

Comment:
another problem i see is that there is currently no range enforcement done here, i.e. calendar_spec_valid() is only called at the very end of parsing, which means we might first create tons of chain elements only to notice later on that most of them are out of date. Or in other words, parsing a specification such as "0..4294967295:00" will result in tons of memory allocations, before we notice that this is not going to work...

A simple way to protect us from this: just compare value with range_end and see if the difference is larger than 230. If so, refuse parsing things here early. Why 230? Because we permit a year range of 1970-2199, and that's the largest range any of our chains can take, with exception of the microsecond chain, but for that one we probably should imply second steps anyway.